File: src/main/java/com/server/ordering/domain/Food.java
package com.server.ordering.domain;

import lombok.*;

import javax.persistence.*;

import static javax.persistence.FetchType.*;
import static javax.persistence.GenerationType.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Food {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "food_id")
    private Long id;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;

    private String foodName;
    private int price;
    private boolean soldOut;
    private String imageUrl;
    private String menuIntro;

    public Food(String foodName, int price, boolean soldOut, String imageUrl, String menuIntro) {
        this.foodName = foodName;
        this.price = price;
        this.soldOut = soldOut;
        this.imageUrl = imageUrl;
        this.menuIntro = menuIntro;
    }

    public void changeSoldOutStatus(Boolean soldOut) {
        this.soldOut = soldOut;
    }

    public void registerRestaurant(Restaurant restaurant) {
        this.restaurant = restaurant;
    }

    public void putFood(String foodName, int price, String menuIntro) {
        this.foodName = foodName;
        this.price = price;
        this.menuIntro = menuIntro;
    }

    public void putFood(String foodName, int price, String menuIntro, String imageUrl) {
        this.foodName = foodName;
        this.price = price;
        this.menuIntro = menuIntro;
        this.imageUrl = imageUrl;
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/Food.kt
package com.byeoru.ordering_server.domain

import javax.persistence.*

@Entity
class Food(foodName: String,
                 price: Int,
                 soldOut: Boolean,
                 imageUrl: String?,
                 menuIntro: String) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "food_id")
    var id: Long? = null
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "restaurant_id")
    var restaurant: Restaurant? = null
        protected set

    var foodName: String = foodName
        protected set

    var price = price
        protected set

    var isSoldOut = soldOut
        protected set

    var imageUrl: String? = imageUrl
        protected set

    var menuIntro: String = menuIntro
        protected set

    fun changeSoldOutStatus(soldOut: Boolean) {
        isSoldOut = soldOut
    }

    fun registerRestaurant(restaurant: Restaurant?) {
        this.restaurant = restaurant
    }

    fun putFood(foodName: String, price: Int, menuIntro: String) {
        this.foodName = foodName
        this.price = price
        this.menuIntro = menuIntro
    }

    fun putFood(foodName: String, price: Int, menuIntro: String, imageUrl: String?) {
        this.foodName = foodName
        this.price = price
        this.menuIntro = menuIntro
        this.imageUrl = imageUrl
    }
}